staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -149,6 +149,21 @@ func NestedSlice(obj map[string]interface{}, fields ...string) ([]interface{}, b
 	return runtime.DeepCopyJSONValue(val).([]interface{}), true, nil
 }
 
+// NestedSliceExists checks for existence of a nested field of []interface{}.
+// Returns (true, nil) if a field with the right type exists, (false, nil) if
+// it does not exist, or (false, error) if an error is encountered traversing
+// obj.
+func NestedSliceExists(obj map[string]interface{}, fields ...string) (bool, error) {
+	val, found, err := nestedFieldNoCopy(obj, fields...)
```

Comment:
Why not just expose this function and let people do what they need to do.  These helpers seem overly restrictive for how you can use a thing.